internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -768,15 +758,21 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
 }
 
 // updateStatus is updating all the AgentClusterInstall Conditions.
-// In case that an error has occured when trying to sync the Spec, the error (syncErr) is presented in SpecSyncedCondition.
+// In case that an error has occurred when trying to sync the Spec, the error (syncErr) is presented in SpecSyncedCondition.
 // Internal bool differentiate between backend server error (internal HTTP 5XX) and user input error (HTTP 4XXX)
 func (r *ClusterDeploymentsReconciler) updateStatus(ctx context.Context, clusterInstall *hiveext.AgentClusterInstall, c *common.Cluster, syncErr error) (ctrl.Result, error) {
 	clusterSpecSynced(clusterInstall, syncErr)
 	if c != nil {
 		clusterInstall.Status.ConnectivityMajorityGroups = c.ConnectivityMajorityGroups
 		if c.Status != nil {
 			status := *c.Status
-			clusterRequirementsMet(clusterInstall, status)
+			registeredHosts, approvedHosts, err := r.getNumOfClusterAgents(ctx, clusterInstall, c)
```

Comment:
maybe this part belong to `clusterRequirementsMet`? after all it is used only for this function.

Reply:
OK i see what you mean, it's because it can fail so it make it look more complicated, you can move it back if you think it will make it clearer.

Reply:
Exactly, though still need to check that the cluster is ready, to avoid redundant fetch of the agents. So I'll try to simplify a bit.